ADBDEV-5273 Use std::nothrow for new allocations

dnskvlnk opened the pull request:
New allocations need to be exception safe, so we replaced regular new operator
by that one that doesn't throw exceptions.

Commit:
Remove redundant null checks

gpdb checks that pointers returned by new operator are not null but that's
redundant since new should throw an exception

BenderArenadata commented:
Allure report https://allure.adsw.io/launch/68123

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1233745

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1233746

whitehawk commented:
Please fix description according to checklist in wiki. For ex, now it is not wrapped to 80 symbols width.

whitehawk commented on gpcontrib/gpcloud/src/gpreader.cpp:
```diff
@@ -33,9 +33,6 @@ unsigned long gpcloud_id_function(void) {
 
 int thread_setup(void) {
     mutex_buf = new pthread_mutex_t[CRYPTO_num_locks()];
```
but what if `new` fails to allocate? I think better to change to `new (std::nothrow)`

RekGRpth commented on gpcontrib/gpcloud/src/gpreader.cpp:
```diff
@@ -102,9 +99,6 @@ GPReader* reader_init(const char* url_with_options) {
         PrepareS3MemContext(params);
 
         reader = new GPReader(params);
-        if (reader == NULL) {
```
https://github.com/arenadata/gpdb/blob/2987c8582ccc878a63b0e69ce2e712e2208810ea/gpcontrib/gpcloud/include/gpreader.h#L50-L51

RekGRpth commented on gpcontrib/gpcloud/src/gpwriter.cpp:
```diff
@@ -82,9 +82,6 @@ GPWriter* writer_init(const char* url_with_options, const char* format) {
 
         string extName = params.isAutoCompress() ? string(format) + ".gz" : format;
         writer = new GPWriter(params, extName);
-        if (writer == NULL) {
```
https://github.com/arenadata/gpdb/blob/2987c8582ccc878a63b0e69ce2e712e2208810ea/gpcontrib/gpcloud/include/gpwriter.h#L51-L52

Commit:
Revert "Remove redundant null checks"

This reverts commit 8e4fcd508eacbc3432348f398fa7353b7d407903.

Commit:
Use std::nothrow for new allocations

New allocations need to be exception safe, so we replaced regular new operator
by that one that doesn't throw exceptions.

RekGRpth commented:
update PR description and put title to it

BenderArenadata commented:
Allure report https://allure.adsw.io/launch/68266

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1239270

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1239271

RekGRpth reviewed (approved):
update PR description and put title to it

whitehawk commented:
Description is still > 80 chars!

Commit:
Merge branch 'adb-6.x-dev' into ADBDEV-5273

BenderArenadata commented:
Allure report https://allure.adsw.io/launch/68540

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1252609
